### ISSUE TITLE
Azure Identity required permission errors are not surfaced.

### ISSUE THREAD

**Issue by sh0rtcircuit** [Azure/azure-sdk-for-python#26123]
- **Package Name**: azure-cosmos
- **Package Version**: 4.3.0
- **Operating System**: Windows 10
- **Python Version**: 3.9

**Describe the bug**
When trying to get the CosmosClient with Azure AAD auth with

```
aad_credential= DefaultAzureCredential()
CosmosClient(url=<my_host_url>, credential=aad_credential)
```

and _not_ having the required AAD rights (i.e., `Microsoft.DocumentDB/databaseAccounts/readMetadata`), an exception that _should_ be surfaced is caught within the library:

Expected surfaced exception:
` azure.cosmos.exceptions.CosmosHttpResponseError: (Forbidden) Request blocked by Auth ********** : Request is blocked because principal [*********] does not have required RBAC permissions to perform action [Microsoft.DocumentDB/databaseAccounts/readMetadata] on resource [/]. Learn more: https://aka.ms/cosmos-native-rbac.
ActivityId: **********, Microsoft.Azure.Documents.Common/2.14.0
Code: Forbidden
`

It is caught in _global_endpoint_manager.py at line 132.

Actual surfaced message (which is not very helpful):
`AttributeError: 'NoneType' object has no attribute 'ConsistencyPolicy'`

**To Reproduce**

```
aad_credential= DefaultAzureCredential()
CosmosClient(url=<my_host_url>, credential=aad_credential)
```

with aad account that does not have `Microsoft.DocumentDB/databaseAccounts/readMetadata` rights on the cosmos resource.

**Expected behavior**
Surface the above described exception.

**Comment by kashifkhan** [comment]
Thanks for the feedback @sh0rtcircuit , we'll investigate asap.

**Comment by EpicWink** [comment]
I suggest having [`_GlobalEndpointManager._GetDatabaseAccount`](https://github.com/Azure/azure-sdk-for-python/blob/c4c060f2a61851983f49e5b01b215d3c8fccde71/sdk/cosmos/azure-cosmos/azure/cosmos/_global_endpoint_manager.py#L116) (and its [async counterpart](https://github.com/Azure/azure-sdk-for-python/blob/014e077bba552d67a23995f52035bafc5a637b01/sdk/cosmos/azure-cosmos/azure/cosmos/aio/_global_endpoint_manager_async.py#L110)) raise the original exception (or use [exception groups](https://pypi.org/project/exceptiongroup/) to raise all exceptions from all regions' attempts) instead of returning [`None`](https://github.com/Azure/azure-sdk-for-python/blob/c4c060f2a61851983f49e5b01b215d3c8fccde71/sdk/cosmos/azure-cosmos/azure/cosmos/_global_endpoint_manager.py#L141).

Looking at the code, none of the consumers of `_GetDatabaseAccount`'s result handle a value of `None` anyway, so this shouldn't break anything (although tests will need to be corrected to handle `CosmosClientConnection` instantiation raising when `consistency_level` is `None`).

There already seems to be some work on this: Azure/azure-sdk-for-python#24234

Possibly relevant: Azure/azure-sdk-for-python#10878

Caused by new feature: Azure/azure-sdk-for-python#21663

**Comment by joshft91** [comment]
Chiming in here as I'm seeing the same error message not with Azure AD, but an IP Firewall setting. Switching CosmosDB public network access from "All networks" to "Selected networks" and whitelisting an IP and then attempting to initialize a Cosmos connection from a forbidden IP results in the `AttributeError: 'NoneType' object has no attribute 'ConsistencyPolicy'` error (in most cases)

Python code when the error is raised:
```python
client = CosmosClient.from_connection_string("XXX")
```

Only once I've seen the following message which outlines which IP was blocked.
```
azure.cosmos.exceptions.CosmosHttpResponseError: (Forbidden) Request originated from IP xxx.xxx.xxx.x through public internet. This is blocked by your Cosmos DB account firewall settings. More info: https://aka.ms/cosmosdb-tsg-forbidden
ActivityId: xxx, Microsoft.Azure.Documents.Common/2.14.0
Code: Forbidden
Message: Request originated from IP xxx.xxx.xxx.x through public internet. This is blocked by your Cosmos DB account firewall settings. More info: https://aka.ms/cosmosdb-tsg-forbidden
```

What would cause the `NoneType object has no attribute 'ConsistencyPolicy'` error vs the more descriptive one above?

**Comment by simorenoh** [comment]
Solved with https://github.com/Azure/azure-sdk-for-python/pull/29256.

**Comment by simorenoh** [comment]
This issue is solved with release 4.4.0b1 and authentication messages should be properly raised now. Thanks!
